src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8261,8 +8348,11 @@ bool tryLookupLengthOrCount(string propertyName, out PropertySymbol? valid, ref
                 {
                     lookupResult.Clear();
                     valid = property;
+                    property.AddUseSiteInfo(ref useSiteInfo);
+                    diagnostics.Add(syntax, useSiteInfo);
```

Comment:
diagnostics.ReportUseSite? Also, shouldn't we report this when we are actually "using" the property. I.e. creating the BoundPropertAccess node? Where do we do this for a regular binding process?